### PR TITLE
convert js to ts

### PR DESCRIPTION
**Pull request by plijyfes**

### CODE REVIEW

File: zk/src/archive/web/js/zk/pkg.ts [hunk 1]
```diff
@@ -14,19 +14,19 @@ it will be useful, but WITHOUT ANY WARRANTY.
...
 zk.copy(zk, (function () {
 	var _loaded = {'zk': true}, //loaded
-		_xloadings = [], //loading (exclude loaded)
-		_loadedsemis = [], //loaded but not inited
-		_afterLoadFronts = [],
-		_afterLoads = [],
+		_xloadings: string[] = [], //loading (exclude loaded)
+		_loadedsemis: string[] = [], //loaded but not inited
+		_afterLoadFronts: {(): void}[] = [],
+		_afterLoads: {(): void}[] = [],
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/pkg.ts:20]: try (() => void)[]

File: zk/src/archive/web/js/zk/zuml/Parser.ts [hunk 2]
```diff
@@ -15,15 +15,15 @@ Copyright (C) 2009 Potix Corporation. All Rights Reserved.
...
+	function _innerText(node: HTMLElement): string | null {
 		var txt = node.innerHTML,
 			j = txt.indexOf('<!--');
 		if (j >= 0)
 			txt = txt.substring(j + 4, txt.lastIndexOf('-->'));
 		txt = txt.trim();
 		return txt ? '<div>' + txt.trim() + '</div>' : null;
 	}
-	function _aftCreate(wgt, cwgts, node, opts) {
+	function _aftCreate(wgt, cwgts, node, opts): zk.Widget | zk.Widget[] | void {
```
Comment (scribetw) [inline zk/src/archive/web/js/zk/zuml/Parser.ts:26]: return null | zk.Widget | zk.Widget[]

**Review by scribetw — CHANGES_REQUESTED** [review]

File: zk/src/archive/web/js/zk/zuml/Parser.ts [hunk 3]
```diff
@@ -78,21 +78,21 @@ Copyright (C) 2009 Potix Corporation. All Rights Reserved.
 			}
 		}
 	}
-	function _create(parent, e, args, cwgts) {
+	function _create(parent, e, args, cwgts): void | null {
```
Comment (plijyfes) [inline zk/src/archive/web/js/zk/zuml/Parser.ts:81]: return void only

**Review by plijyfes — COMMENTED** [review]
add es2015.promise to tsconfig

**Review by DevChu — APPROVED** [review]

**Review by scribetw — APPROVED** [review]

**Pull request merged**